graphics/palette.cpp

```diff
@@ -83,6 +83,28 @@ bool Palette::contains(const Palette& p) const {
 	return p._size <= _size && !memcmp(_data, p._data, p._size * 3);
 }
 
+void Palette::clear() {
+	delete[] _data;
+	_data = nullptr;
+	_size = 0;
+}
+
+void Palette::resize(uint newSize) {
+	if (newSize > _size)
+	{
+		byte *newData = nullptr;
+		if (newSize > 0) {
+			newData = new byte[newSize * 3]();
```

Comment:
It may be interesting to switch to malloc/free and use realloc here.

Reply:
Realloc would handle a copy for us. We could look into switching, but it feels a bit outside the scope of this PR.  
I'm not worried about optimizing out an extra copy though; resizing a palette should be a rare occurrence, especially from anything than of than empty palette.

Reply:
So adding a flag to ask for preserving the contents could avoid a useless copy

Reply:
Ok, perserve flag is added.